tests/test_avgHR.py

```diff
@@ -35,3 +44,29 @@ def test_avghr():
     assert abs(avghr(times1, voltages1,   lower_bound1,  times1[len(times1)-1]) - hrlist[0]) < 5
     assert abs(avghr(times2, voltages2,   lower_bound2,  times2[len(times2)-1]) - hrlist[1]) < 5
     assert abs(avghr(times3, voltages3,   lower_bound3,  times3[len(times3)-1]) - hrlist[2]) < 5
+
+    assert abs(HRM1.averagehr - hrlist[0]) < 5
+    assert abs(HRM2.averagehr - hrlist[1]) < 5
+    assert abs(HRM3.averagehr - hrlist[2]) < 5
+
+    lower_bound1 = times1[0] + 0.5 * (times1[len(times1) - 1] - times1[0])
+    lower_bound2 = times2[0] + 0.5 * (times2[len(times2) - 1] - times2[0])
+    lower_bound3 = times3[0] + 0.5 * (times3[len(times3) - 1] - times3[0])
+
+    HRM1.t1 = lower_bound1
+    HRM2.t1 = lower_bound2
+    HRM3.t1 = lower_bound3
+
+    HRM1.t2 = times1[len(times1)-1]
+    HRM2.t2 = times2[len(times2)-1]
+    HRM3.t2 = times3[len(times3)-1]
+
+    HRM1.avghr()
+    HRM2.avghr()
+    HRM3.avghr()
+
+    assert abs(HRM1.averagehr - hrlist[0]) < 5
+    assert abs(HRM2.averagehr - hrlist[1]) < 5
+    assert abs(HRM3.averagehr - hrlist[2]) < 5
+
```

Comment:
some comment about the replicated lines for the 3 test cases; consider doing in a loop